src/DeepLinkModule/javasource/deeplink/actions/StartDeeplinkJava.java

```diff
@@ -200,28 +203,30 @@ else if (!deeplink.getAllowGuests().booleanValue() && session.getUser().isAnonym
 			
 			//switch to the users context
 			context = session.createContext(); 
-			IUser sessionUserObj = session.getUser();
+			IUser sessionUserObj = session.getUser(sessionContext);
 			String userName = sessionUserObj.getName();
 			
 			// Retrieve the language set for the deeplink.
 			Language language = deeplink.getDeepLink_Language();
 			if (language != null) {
 				// Deeplink has a language, put it on the session user.
-				User user = User.initialize(context, sessionUserObj.getMendixObject());
+				User user = User.initialize(sessionContext, sessionUserObj.getMendixObject());

```

Comment:
It seems we no longer use `context` in the rest of the method, so:

* we should be able to remove line 204, which switches the context to the user context. It has no functionality any more and feels confusing.
* we could rename it to the `systemContext` to make it explicit, since we also have a `sessionContext` (which could also named as `userContext` to be aligned)